client.go

```diff
@@ -864,6 +859,14 @@ func getRedirectURL(baseURL string, location []byte) string {
 	return redirectURL
 }
 
+func IsFinishStatusCode(statusCode int) bool {
```

Comment:
Can you add some documentation to this function. Also a better name might be `StatusCodeIsRedirect`?

Reply:
Done. Check please.